product_extended_variants/model/template.py

```diff
@@ -117,7 +119,7 @@ def do_change_standard_price(self, cr, uid, ids, new_price, context=None):
 
                 for prod_variant in product.product_variant_ids:
                     qty = prod_variant.qty_available
-                    if prod_variant.cost_method != 'standard' or not qty:
+                    if not qty or make_journal_entry:
```

Comment:
This condition should be like:
`if prod_variant.cost_method != 'standard' and not make_journal_entry or not qty:`

Otherwise it will be applied to all Products regardless of the Cost Method.

Best Regards